pollux/lib/core/src/main/scala/io/iohk/atala/pollux/core/service/PresentationServiceImpl.scala

```diff
@@ -401,7 +440,96 @@ private class PresentationServiceImpl(
             }
         )
     } yield presentationPayload
+  }
 
+  private def createAnoncredPresentationPayloadFromCredential(
+      issuedCredentials: Seq[IssueCredential],
+      schemaIds: Seq[String],
+      credentialDefinitionIds: Seq[UUID],
+      requestPresentation: RequestPresentation,
+  ): ZIO[WalletAccessContext, PresentationError, AnoncredPresentation] = {
+    for {
+      schemaMap <-
+        ZIO
+          .collectAll(schemaIds.map { schemaId =>
+            for {
+              uri <- ZIO.attempt(new URI(schemaId)).mapError(e => UnexpectedError(e.getMessage))
+              content <- uriDereferencer.dereference(uri).mapError(e => UnexpectedError(e.error))
+              vcSchema <- parseCredentialSchema(content).mapError(e => UnexpectedError(e.message))
+              anoncredSchema <- AnoncredSchemaSerDesV1.schemaSerDes
+                .deserialize(vcSchema.schema)
+                .mapError(e => UnexpectedError(e.error))
+              anoncredLibSchema =
+                SchemaDef(
+                  schemaId,
+                  anoncredSchema.version,
+                  anoncredSchema.attrNames,
+                  anoncredSchema.issuerId
+                )
+            } yield (schemaId, anoncredLibSchema)
+          })
+          .map(_.toMap)
```

Comment:
I would recommend extracting this logic into the method. Something like `resolveSchema(schemaId)`

Reply:
fixed